addons/sourcemod/scripting/scp_sf/stocks.sp

```diff
@@ -1694,4 +1694,64 @@ stock int TraceClientViewEntity(int client)
 public bool TRDontHitSelf(int entity, int mask, any data)
 {
 	return (1 <= entity <= MaxClients) && (entity != data);
+}
+
+stock int TF2_CreateLightEntity(float radius, int color[4], int brightness, float lifetime)
+{
+	int entity = CreateEntityByName("light_dynamic");
+	if (entity != -1)
+	{			
+		char lightColor[32];
+		Format(lightColor, sizeof(lightColor), "%d %d %d", color[0], color[1], color[2]);
+		DispatchKeyValue(entity, "rendercolor", lightColor);
+		
+		SetVariantFloat(radius);
+		AcceptEntityInput(entity, "spotlight_radius");
+		
+		SetVariantFloat(radius);
+		AcceptEntityInput(entity, "distance");
+		
+		SetVariantInt(brightness);
+		AcceptEntityInput(entity, "brightness");
+		
+		SetVariantInt(1);
+		AcceptEntityInput(entity, "cone");
+		
+		DispatchSpawn(entity);
+		
+		ActivateEntity(entity);
+		AcceptEntityInput(entity, "TurnOn");
+		SetEntityRenderFx(entity, RENDERFX_SOLID_SLOW);
+		SetEntityRenderColor(entity, color[0], color[1], color[2], color[3]);
+		
+		int flags = GetEdictFlags(entity);
+		if (!(flags & FL_EDICT_ALWAYS))
+		{
+			flags |= FL_EDICT_ALWAYS;
+			SetEdictFlags(entity, flags);
+		}
+		
+		CreateTimer(lifetime, Timer_DestroyLight, EntIndexToEntRef(entity), TIMER_FLAG_NO_MAPCHANGE);	
+	}
+	
+	return entity;
+}
+
+public Action Timer_DestroyLight(Handle timer, int ref)
+{
+	int entity = EntRefToEntIndex(ref);
+	if (entity > MaxClients)
+	{
+		AcceptEntityInput(entity, "TurnOff");
+		RequestFrame(Frame_KillLight, ref);
```

Comment:
Is this necessary? I'd expect the entity to turn itself off and clean up after itself on removal.

Reply:
Sorry, wrong line. But I do question whether it's necessary to turn off the entity, and also to delay the removal.

Reply:
looks like it is not, I took the light stock from VSH and it was accompanied with this light-specific timer and delay, while also having a separate entity cleanup timer. Just thought they exist for a reason and never questioned it lol